### PR TITLE
ban userr from MPOS, submit worker current diff, correct auto create worker

### PR DESCRIPTION
**Pull request by cornac**

### CODE REVIEW

File: Database/MySQLMPOS.py [hunk 1]
```diff
@@ -18,46 +18,55 @@ def get_uid(self, username):
...
 			return False
+		   elif	row['is_locked'] == 2:
+			self.logger.error("User {} is banned by admin".format(user))
+			return False
 		   else:
 			return row['id']
 		   return False
-	    except:
-		self.logger.error(e)
+	    except Exception :
```
Comment (ahmedbodi) [inline Database/MySQLMPOS.py]:
```suggestion
	    except Exception as exc:
	        self.logger.error(exc)
```

File: Database/MySQLMPOS.py [hunk 2]
```diff
@@ -18,46 +18,55 @@ def get_uid(self, username):
...
 		return False
 
 	def authorise(self, *args, **kwargs):
 	    self.logger.info("Checking Auth For User: {0}".format(kwargs['username']))
 	    connection = self.connect()
-	    try:
+	    uid = self.get_uid(kwargs['username'])
```
Comment (ahmedbodi) [inline Database/MySQLMPOS.py:40]:
```suggestion
            if not 'username' in kwargs:
                return 0
	    uid = self.get_uid(kwargs['username'])
```

File: Database/MySQLMPOS.py [hunk 3]
```diff
@@ -18,46 +18,55 @@ def get_uid(self, username):
...
+	    if uid:
+	     try:
 		with connection.cursor() as cursor:
-		   cursor.execute("SELECT id FROM `pool_worker` WHERE `username` = %(username)s", kwargs)
+		   cursor.execute("SELECT `id` FROM `pool_worker` WHERE `username` = %(username)s", kwargs)
 		   result = cursor.fetchone()
-		   if result is None and not kwarg['create_user']:
-			return 0
+		   self.logger.info(kwargs)
```
Comment (ahmedbodi) [inline Database/MySQLMPOS.py]:
```suggestion
```
Remove Logline please

File: Database/MySQLMPOS.py [hunk 4]
```diff
@@ -18,46 +18,55 @@ def get_uid(self, username):
...
+		   if result is None and not kwargs['create_user']:
+			return 0 
 		   elif result is None and kwargs['create_user']:
-			uid = self.get_uid(kwargs['username'])
```
Comment (ahmedbodi) [inline Database/MySQLMPOS.py:42]: What's going on here?

Comment (ahmedbodi) [reply Database/MySQLMPOS.py:42]: This would return False, therefore not creating the user that doesnt exist

File: Database/MySQLMPOS.py [hunk 5]
```diff
@@ -18,46 +18,55 @@ def get_uid(self, username):
...
-	    return False
+	     return False
 
         def insert_shares(self, *args, **kwargs):
 	    kwargs['lres'] = 'Y' if kwargs['lres'] == True else 'N'
             connection = self.connect()
             try:
                 with connection.cursor() as cursor:
-            	   cursor.execute("""INSERT INTO `shares` (time, rem_host, username, our_result, upstream_result, reason, solution, difficulty)
+                   cursor.execute("SELECT `id` FROM `pool_worker` WHERE `username` = %(uname)s", kwargs)
```
Comment (ahmedbodi) [inline Database/MySQLMPOS.py:62]: Why are you checking for the pool worker in insert_shares? At this stage the user should already be authorised and so the user should exist.

File: Database/MySQLMPOS.py [hunk 6]
```diff
@@ -18,46 +18,55 @@ def get_uid(self, username):
...
+                   result = cursor.fetchone()
+                   if result is None    :
+			uid = self.get_uid(kwargs['uname'])
+                        query = "INSERT INTO pool_worker (`account_id`, `username`, `password`) VALUES ({}, '{}', 'x');".format(uid, kwargs['uname'])
```
Comment (ahmedbodi) [inline Database/MySQLMPOS.py:66]: Dont hardcode the worker's password. Some people have custom difficulty and other password pased parameters

**Review by ahmedbodi — CHANGES_REQUESTED** [review]
Fix changes

File: Database/MySQLMPOS.py (hunk 5 above)
Comment (cornac) [reply Database/MySQLMPOS.py:62]: because if user have more than one worker, then worker not been create

File: Database/MySQLMPOS.py (hunk 6 above)
Comment (cornac) [reply Database/MySQLMPOS.py:66]: it was your hardcode)))

File: Database/MySQLMPOS.py (hunk 5 above)
Comment (cornac) [reply Database/MySQLMPOS.py:62]: And user authorize if created user in accounts table. Each user can create not only one worker.

**Pull request closed without merging**